modules/post/linux/manage/xdg_open.rb

```diff
@@ -0,0 +1,33 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Post
+  include Msf::Post::Linux::System
+
+  def initialize(info={})
+    super( update_info( info,
+      'Name'          => 'Linux Admin XDG open',
+      'Description'   => %q{
+          This module will open any local resource in the target system via the 'xdg-open' command.
+      },
+      'License'       => MSF_LICENSE,
+      'Author'        => ['DeveloppSoft'],
+      'Platform'      => ['linux'],
+      'SessionTypes'  => ['shell', 'meterpreter']
+    ))
+
+    register_options([
+      OptString.new('RES', [true, 'Resource to open, URL or file.'])
+    ])
+  end
+
+  def run
+    unless command_exists? 'xdg-open'
```

Comment:
this might be nicer as a multi module that can call the correct method of opening a file on any platform, e.g. gnome-open on other systems, open(1) on macOS, one of the many methods on Windows https://superuser.com/questions/246825/open-file-from-the-command-line-on-windows

Reply:
Indeed, will look at that